### ISSUE TITLE
[Needs Information] unable to drink from leather flask with shield

### ISSUE THREAD

**Issue by schindlershadow** [wendall911/Homeostatic#70]
unable to drink from leather flask with shield or anything in offhand with a rightclick ability

**Comment by wendall911** [comment]
Yeah, unfortunately this is a Minecraft and Forge issue. I've been aware of this for a while. I will have to figure out how to force this to work correctly. I probably need to do a custom Mixin to resolve this. I will work on this tomorrow and give some feedback. This has been an issue since 1.18.2 unfortunately.

**Comment by wendall911** [comment]
This is working correctly in 1.21.1. I'm doing some backporting today. I'll see if I can resolve this.

**Comment by wendall911** [comment]
Wait, this is working in 1.20.1 and 1.21.1 ... what version are you seeing this issue?

**Comment by schindlershadow** [comment]
im in 1.21.1 

trying it will vanilla shield:
![2024-10-04_15 03 17](https://github.com/user-attachments/assets/99f765db-2332-4587-b7b6-59f2d9252921)

**Comment by wendall911** [comment]
Ok, I see the issue on one of our packs. Let me do some investigation and see what mod might be causing this. It is definitely not a Homeostatic issue, but I want to figure out what the cause is. I might be able to force it to work.

**Comment by wendall911** [comment]
Ok, I have no idea why this is happening. I will do more research and get a fix applied.

**Comment by Jadan1213** [comment]
I was just coming to mention this and i see this open. What I've noticed is that it works, but not properly. With a shield and flask equipped, when using, it won't make any sound like you're drinking. However, if I hold the right click button long enough, it WILL apply the water to the thirst bar.

I'm also on 1.20.1

**Comment by wendall911** [comment]
@Jadan1213 Are you using the latest version? This hasn't been an issue for a very long time. I am completely unable to reproduce this.

**Comment by wendall911** [comment]
I think it would be useful is someone could produce a mod list. I think there is another mod messing with this mechanic. I'd be curious to know if this also happens with food or other drinkable items.

**Comment by Jadan1213** [comment]
> [@Jadan1213](https://github.com/Jadan1213) Are you using the latest version? This hasn't been an issue for a very long time. I am completely unable to reproduce this.

I am, yes.

**Comment by Jadan1213** [comment]
> I think it would be useful is someone could produce a mod list. I think there is another mod messing with this mechanic. I'd be curious to know if this also happens with food or other drinkable items.

I'll try to figure out as best I can what combination of mods is causing this.

**Comment by wendall911** [comment]
Yes, you can just give yourself the thirst effect.
/effect give {your_player_name} homeostatic:thirst 150 5 true

If fully hydrated, should leave you with about 3 water droplets of hydration.

**Comment by Jadan1213** [comment]
After testing all day here's what i've got.

single player
--------------
Homeostatic by itself : works
Slowly adding all mods from my client modpack into an instance : works
Adding all server modpack mods into client instance : works

Multiplayer
------------
Same mods. Client modpack instance, connecting to local Server modpack instance : doesn't work properly.


All the same mods all the same config. the only difference is single player VS Server.

Edit: One thing of note! The drinking sound WILL play for OTHER players that are near you. Just not for yourself. That's what originally tipped me off to it working but not properly.

**Comment by wendall911** [comment]
Oh, I see. Let me work on this, as I'm not doing anything unusual. It may be that I need to add a component for Minecraft to understand that it is a consumable.

**Comment by Jadan1213** [comment]
Can I assume this fix will be backported to 1.20.1 as well?

**Comment by wendall911** [comment]
@Jadan1213 Working on something. 1.20.1 is a bit of a pain to fix. I'll have something out tomorrow.

**Comment by wendall911** [comment]
@Jadan1213 Interestingly, if you open a single player world before joining a server, it will work correctly. This is because I'm not correctly syncing data for drinkable items to the client.

**Comment by Jadan1213** [comment]
> [@Jadan1213](https://github.com/Jadan1213) Interestingly, if you open a single player world before joining a server, it will work correctly. This is because I'm not correctly syncing data for drinkable items to the client.

Interesting. Well I'm glad I could help with finding the cause =D. I appreciate the fix!

**Comment by wendall911** [comment]
@Jadan1213 I just updated 1.20.1 on CurseForge and Modrinth. I will release 1.21.1 and 1.21.10 later today. I am currently working on additional features for those versions.

**Comment by wendall911** [comment]
I will close this once published.
